### PR TITLE
Improve DoB detector

### PR DESCRIPTION
**Pull request by thomasbird**
* Dont search line by line for dates, search the whole doc
* Fix context searching, previously the indices `i_line - self.context_before` could go negative, searching the wrong part of the text
* Add more test cases

### CODE REVIEW

File: tests/test_detector_date_of_birth.py [hunk 1]
```diff
@@ -38,18 +38,48 @@ def test_DoB_3(self):
...
         BEFORE: The claimant's, d.o.b. is 4 June 1976
         AFTER:  The claimant's, d.o.b. is {{DATE_OF_BIRTH}}
         """
         self.compare_before_after()
 
-    def test_DoB_5(self):
+    def test_DoB_9(self):
         """
         BEFORE: 1985-01-01 is my birthday.
         AFTER:  {{DATE_OF_BIRTH}} is my birthday.
```
Comment (mynameisbasit) [inline tests/test_detector_date_of_birth.py:85]: this isn't necessary but it would be good to have a test that covers the context feature

Comment (thomasbird) [reply tests/test_detector_date_of_birth.py:85]: Agreed...

File: scrubadub/detectors/date_of_birth.py [hunk 2]
```diff
@@ -103,49 +103,61 @@ def iter_filth(self, text: str, document_name: Optional[str] = None) -> Generato
         :return: The found Filth in the text
         :rtype: Generator[Filth]
         """
+
+        # using the dateparser lib - locale can be set here
+        try:
+            date_picker = search_dates(text, languages=[self.language])
+        except RecursionError:
+            logger = logging.getLogger("scrubadub.detectors.date_of_birth.DateOfBirthDetector")
+            logger.error(f"The documnet '{document_name}' caused a recursion error in dateparser.")
```
Comment (mynameisbasit) [inline scrubadub/detectors/date_of_birth.py:109]: Type on document

Comment (thomasbird) [reply scrubadub/detectors/date_of_birth.py]: fixed

**Pull request merged**